desktopexporter/app/components/sidebar-view/trace-list.tsx

```diff
@@ -144,18 +145,50 @@ type TraceListProps = {
 export function TraceList(props: TraceListProps) {
   let ref = useRef(null);
   let size = useSize(ref);
+
   let location = useLocation();
+  let navigate = useNavigate();
+
+  let selectedIndex = 0;
+  let selectedTraceID = "";
   let { traceSummaries } = props;
 
   // Default to the first trace in the list if none are selected
-  let selectedTraceID = "";
   if (location.pathname.includes("/traces/")) {
     selectedTraceID = location.pathname.split("/")[2];
+    selectedIndex = traceSummaries.findIndex(
+      (summary) => summary.traceID === selectedTraceID,
+    );
   } else {
-    selectedTraceID = traceSummaries[0].traceID;
+    selectedTraceID = traceSummaries[selectedIndex].traceID;
     window.location.href = `/traces/${selectedTraceID}`;
   }
 
+  // Set up keyboard navigation
+  let arrowLeftPressed = useKeyPress("ArrowLeft");
+  let arrowRightPressed = useKeyPress("ArrowRight");
+  let hPressed = useKeyPress("h");
+  let lPressed = useKeyPress("l");
+
+  useEffect(() => {
+    if (arrowLeftPressed || hPressed) {
+      selectedIndex = selectedIndex > 0 ? selectedIndex - 1 : 0;
+      selectedTraceID = traceSummaries[selectedIndex].traceID;
+      navigate(`/traces/${selectedTraceID}`);
+    }
+  }, [arrowLeftPressed, hPressed]);
+
+  useEffect(() => {
+    if (arrowRightPressed || lPressed) {
+      selectedIndex =
+        selectedIndex < traceSummaries.length - 1
+          ? selectedIndex + 1
+          : traceSummaries.length - 1;
+      selectedTraceID = traceSummaries[selectedIndex].traceID;
+      navigate(`/traces/${selectedTraceID}`);
+    }
```

Comment:
These chunks of code are a little mysterious without a comment or a name. Factoring them out into a named function might make this read a bit better

```typescript

if (arrowLeftPressed || hPressed) {
      navigateToPreviousTrace()
}

if (arrowRightPressed || lPressed) {
  navigateToNextTrace()
}
```